### ISSUE TITLE
Error C2338 when adding an std::map<int, std::string> option

### ISSUE THREAD

**Issue by dmf254** [CLIUtils/CLI11#610]
Greetings,

When trying to add the following option, I am greeted with an error:

```C
std::map<int, std::string> userid_map;
app.add_option("--userid-map", userid_map, "A map of userid->name key values");
```
```
C:\...\CLI11.hpp(1586,38): error C2338: option object type must have a lexical cast overload or streaming input operator(>>) defined, if it is convertible from another type use the add_option<T, XC>(...) with XC being the known type
```

Any ideas? Also, what would be the .ini format for this kind of container? Thanks!

**Comment by phlptp** [comment]
Are you on the latest master or one of the releases.  That may only work on the master at the moment.  

also for the ini file I would probably do something like

app.add_option("--userid-map", userid_map, "A map of userid->name key values")->delimiter(',');

then in the ini files you just enter 
userid-map=4,user

actually not sure what it would look like on the output,  I would guess it generates a vector of some kind

**Comment by dmf254** [comment]
I'm currently using the latest release (1.9.1). I was going off of the documentation where it discusses "Containers of options" and lists map/unordered_map and assumed that it would work. A ```vector<pair<int, string>>``` container works as intended and I think that will be okay for my use case. Would the ini format be similar for a ```vector<pair<int,string>>```?

**Comment by phlptp** [comment]
I think there was a bug for some more complex containers that got fixed after 1.9.0 and would be in the 2.0 release.  Whereas vectors of pair/tuple support was added in 1,9.    And yeah I think the ini format would be the same for vector<pair<..>>

**Comment by henryiii** [comment]
2.0 should come out this weekend, roughy. This week is PyHEP so I'm not very available if something had broken. (And next week is SciPy...)
